crates/common/src/config.rs

```diff
@@ -96,54 +100,59 @@ impl Config {
     }
 }
 
-#[derive(Debug, Default, Clone)]
+#[derive(Debug, Default, Clone, PartialEq, Eq, Hash)]
 pub struct IngotMetadata {
     pub name: Option<SmolStr>,
     pub version: Option<Version>,
 }
 
-#[derive(Debug, Clone)]
-pub enum DependencyDescription {
-    Path(Utf8PathBuf),
-    PathWithArguments {
-        path: Utf8PathBuf,
-        arguments: IngotArguments,
-    },
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct DependencyDescription {
+    pub url: Url,
```

Comment:
This might not be expressive enough to handle both relative and absolute dependency paths.